### DISCUSSION TITLE
Public Transport evaluation

### DISCUSSION THREAD

**Discussion by kiselev-dv** [organicmaps/organicmaps#8334] [Ideas]
# GTFS

* Evaluate GTFS availability
* GTFS data deduplication
  * How often trips vary for the route
  * Geometry artifacts for trip/route geometry
     * are stops always within some distance form route geometry
     * do we have geometry tails for route
  * different transit aggregators giving the same route but with different details

# OSM data related to GTFS

OSM taginfo for GTFS: https://taginfo.openstreetmap.org/search?q=gtfs#keys

* Check how well data from OSM https://taginfo.openstreetmap.org/keys/gtfs_id#map matches with GTFS

# PT Routing algorithms / libraries

# GTFS aggregators
* transitland

**Comment by kiselev-dv** [comment]
## Singapour

* Use their own format JSON/XML
* Data available at https://datamall.lta.gov.sg/content/datamall/en/dynamic-data.html#Public%20Transport
* OSM data use `ref` and `route_ref` on stops

## Canada / New Brunswick

* GTFS feed: https://open.moncton.ca/documents/moncton::transit-files-gtfs/about
* OSM data use `gtfs_id` on stops, ids are AlphaNum

## Canada / Toronto

* GTFS: https://www.transit.land/places/Canada/Ontario/Toronto
* Different providers
* OSM data: Some stops have `ref` matching gtfs code, also there are `ref:<transit operator>`

**Comment by kiselev-dv** [comment]
Overpass turbo link for PT platforms https://overpass-turbo.eu/s/1Mhf

**Comment by cjj20** [comment]
### Cairns, Queensland, Australia 
Gtfs source: https://www.data.qld.gov.au/dataset/general-transit-feed-specification-gtfs-translink/resource/9850f8e5-f572-4de7-97c5-84d45526f9bd
Gtfs source for Australia: https://translink.com.au/about-translink/open-data/gtfs-rt

- osm tags use _ref_

**Comment by cjj20** [comment]
### China 
Gtfs source: [mapabc.com](https://www.mapabc.com/#/)
(the datasets may need to be requested and purchased)
Alternate gtfs source: Tianditu - state mapping service open to persons as well as companies, unsure of what formats available or if open to non-citizens; 
Baidu - mapping tool/app but unsure if gives out datasets

**Comment by cjj20** [comment]
### Hong Kong, China
Gtfs source: https://data.gov.hk/en-datasets
https://data.gov.hk/en-data/dataset/hk-td-tis_11-pt-headway-en

-  the data did not match so but osm tags are  _ref_, _local_ref_, _ref:<operator_

**Comment by cjj20** [comment]
### Rome, Italy
Gtfs source: https://romamobilita.it/en/technology/open-data/dataset leads to zip file only containing a readme. The readme file gives actual link for gtfs data at: http://dati.muovi.roma.it/gtfs/rome_static_gtfs.zip

- osm _ref_ used to match to gtfs

**Comment by cjj20** [comment]
### Italy
Gtfs source: https://dati.gov.it/view-dataset (has datasets for different cities which are from openstreetmap sources but are in gtfs)

**Comment by cjj20** [comment]
### Barcelona, Spain
Gtfs source: AMB Bus (bus operator)  gtfs  https://nap.transportes.gob.es/Files/Detail/1295

- osm tags _ref_, _local_ref_(local_ref only matched two stops), _route_ref_ (route_ref had no matches)

Other gtfs source: TMB Bus and Metro Barcelona  https://nap.transportes.gob.es/Files/Detail/1127

- osm tags _ref_, _route_ref_(route_ref only matched two stops)

**Comment by cjj20** [comment]
### Madrid, Spain
Gtfs source: CRTM Metro https://nap.transportes.gob.es/Files/Detail/933

- osm tag _ref_ (matched only 1 stop, this dataset had very few ref tags)

### Spain
Gtfs source: https://nap.transportes.gob.es/Files/List?search=&orderby=Relevancia 
(gtfs datasets seem to be separated into operators)

**Comment by cjj20** [comment]
### Hyderabad, Telgana, India 
Gtfs source: https://data.telangana.gov.in/dataset/gtfs-mmts-hyderabad

- osm tag _ref_ doesn't match

### Delhi, India 
Gtfs source: Delhi bus gtfs https://otd.delhi.gov.in/data/static/

- osm data doesn't have any ref, so no matches

### Delhi, India
Gtfs source: DMRC metro gtfs https://otd.delhi.gov.in/data/staticDMRC/

- osm data doesn't have any ref, so no matches

**Comment by hb0nd** [comment]
there's also this list on the OSM wiki: https://wiki.openstreetmap.org/wiki/List_of_GTFS_feeds
might be good to copy what we learn on this thread to it, for future people :)

**Reply by cjj20** [reply]
> there's also this list on the OSM wiki: https://wiki.openstreetmap.org/wiki/List_of_GTFS_feeds might be good to copy what we learn on this thread to it, for future people :)

Thank you, I agree and I will update the osm wiki.

**Comment by cjj20** [comment]
### Poznan, Poland 
Gtfs source: ZTM Poznan https://www.ztm.poznan.pl/pl/dla-deweloperow/gtfsFiles  
https://www.ztm.poznan.pl/pl/dla-deweloperow/getGTFSFile

- osm tag _ref_, _old_ref_(_old_ref_ had very few matches)

### Szczecin, Poland
Gtfs source: https://www.zditm.szczecin.pl/pl/zditm/dla-programistow/gtfs

- osm tags _ref_ and _local_ref_ but none match

### Gdansk, Poland
Gtfs source: https://ckan.multimediagdansk.pl/dataset/tristar

- osm tags _ref_, _ref:ztm_
